pkg/unixserver/server.go

```diff
@@ -257,16 +365,21 @@ func (a *CDMetrics) Describe(ch chan<- *prometheus.Desc) {
 	}
 }
 
-//Collect implements prometheus.Collector.
+//Collect implements prometheus.Collector
 func (a *CDMetrics) Collect(ch chan<- prometheus.Metric) {
+	a.mu.RLock()
+	defer a.mu.RUnlock()
 	for _, metric := range a.metrics {
-		for _, labeled_metric := range metric {
-			ch <- prometheus.NewMetricWithTimestamp(labeled_metric.timeStamp, prometheus.MustNewConstMetric(labeled_metric.metricDesc, labeled_metric.valueType, labeled_metric.metric,
-				labeled_metric.host, labeled_metric.pluginInstance, labeled_metric.typeInstance))
+		metric.mu.RLock()
+		defer metric.mu.RUnlock()
+		for _, labeledMetric := range metric.labels {
+			ch <- prometheus.NewMetricWithTimestamp(labeledMetric.timeStamp, prometheus.MustNewConstMetric(labeledMetric.metricDesc, labeledMetric.valueType, labeledMetric.metric,
```

Comment:
I feel like this should be showing up as a conflict with this code: https://github.com/infrawatch/sg2/pull/5/files#diff-aa02aac35c431b4ae0e6c9958706f53aR266

OH! I see this PR is targeted to merge against csibbitt-756-compatibility-fixes, but that branch is already fully merged (and 10 commits behind) master. Can you re-target this PR to be against master, please? The sg2 isn't use in anything yet, so we're not accumulating a long lived topic branch here.

Reply:
yeah will do